src/cpu/slime_check/simd.zig

```diff
@@ -0,0 +1,151 @@
+const std = @import("std");
+
+/// This is experimentally faster than the value std.simd.suggestVectorLength gives, which is 4
+pub const lanes = 8;
+
+pub const Vec64 = @Vector(lanes, i64);
+pub const Vec32 = @Vector(lanes, i32);
+pub const Vecb = @Vector(lanes, bool);
+
+/// Tests whether the chunks [x, y]..(x, z + lanes) are slime chunks
+pub fn areSlime(world_seed: i64, x: i32, z: i32) Vecb {
+    var random = Random.init(getRandomSeeds(world_seed, x, z));
+    return random.nextInts(10) == @as(Vec32, @splat(0));
+}
+
+/// Tests whether the chunks [x, y]..(x, z + lanes) are slime chunks
+/// Uses a biased random function that is faster but very occasionally
+/// (<1 in 100,000,000) gives an incorrect result
+pub fn areSlimeBiased(world_seed: i64, x: i32, z: i32) Vecb {
+    var random = Random.init(getRandomSeeds(world_seed, x, z));
+    return random.nextIntsBiased(10) == @as(Vec32, @splat(0));
+}
+
+/// Returns the seeds used by the PRNG for chunks [x, z]..(x, z + lanes)
+/// See scalar.zig
+pub fn getRandomSeeds(world_seed: i64, x: i32, z: i32) Vec64 {
+    const world_seeds: Vec64 = @splat(world_seed);
+
+    const x_increment: Vec64 = @splat(
+        @as(i64, x *% x *% 4987142) +%
+            @as(i64, x *% 5947611),
+    );
+
+    const magic1: Vec64 = @splat(4392871);
+    const magic2: Vec32 = @splat(389711);
+
+    const zs: Vec32 = risingSequence(z);
+    const z_increment =
+        @as(Vec64, zs *% zs) *% magic1 +%
+        @as(Vec64, zs *% magic2);
+
+    const magic3: Vec64 = @splat(987234911);
+    return world_seeds +% x_increment +% z_increment ^ magic3;
+}
+
+pub fn risingSequence(n: i32) Vec32 {
+    return @as(Vec32, @splat(n)) + comptime blk: {
+        var vec: Vec32 = @splat(0);
+        for (0..lanes) |i| vec[i] = i;
+        break :blk vec;
+    };
+}

```

Comment:
this can probably just be inlined at the callsite

also is this actually faster than the naive impl?
```zig
var vec: Vec32 = undefined;
for (0..lanes) |i| vec[i] = @intCast(i + n);
return vec;
```

Reply:
> this can probably just be inlined at the callsite
it's mostly a relic from when i was testing different vector lengths, i'll do that now